src/Development/IDE/Core/PositionMapping.hs

```diff
@@ -29,20 +46,33 @@ fromCurrentRange :: PositionMapping -> Range -> Maybe Range
 fromCurrentRange mapping (Range a b) =
     Range <$> fromCurrentPosition mapping a <*> fromCurrentPosition mapping b
 
-idMapping :: PositionMapping
-idMapping = PositionMapping Just Just
+zeroMapping :: PositionMapping
+zeroMapping = PositionMapping idDelta
 
 -- | Compose two position mappings. Composes in the same way as function
 -- composition (ie the second argument is applyed to the position first).
-composeMapping :: PositionMapping -> PositionMapping -> PositionMapping
-composeMapping (PositionMapping to1 from1) (PositionMapping to2 from2) =
-  PositionMapping (to1 <=< to2)
-                  (from1 >=> from2)
-
-applyChange :: PositionMapping -> TextDocumentContentChangeEvent -> PositionMapping
-applyChange PositionMapping{..} (TextDocumentContentChangeEvent (Just r) _ t) = PositionMapping
-    { toCurrentPosition = \x -> toCurrent r t =<< toCurrentPosition x
-    , fromCurrentPosition = \x -> fromCurrentPosition  =<< fromCurrent r t x
+composeDelta :: PositionDelta
+                -> PositionDelta
+                -> PositionDelta
+composeDelta (PositionDelta to1 from1) (PositionDelta to2 from2) =
+  PositionDelta (to1 <=< to2)
+                (from1 >=> from2)
+
+idDelta :: PositionDelta
+idDelta = PositionDelta Just Just
```

Comment:
Why not define `Monoid` and `Semigroup` instances for `PositionDelta`?

Reply:
It didn't occur to me in particular. I'm a GHC developer so I prefer having explicitly named functions!

Reply:
Lies. If you were a real GHC developer you'd have used `;` `{` and your commit would have been rendered in comic sans.